src/Model/ModelManager.php

```diff
@@ -217,18 +218,19 @@ public function executeQuery(object $query)
             return $query->getQuery()->execute();
         }
 
-        if ($query instanceof ProxyQuery) {
+        if ($query instanceof Query || $query instanceof ProxyQuery) {
```

Comment:
Please use a different if
```suggestion
        if ($query instanceof Query) {
            return $query->execute();
        }

        if ($query instanceof ProxyQuery) {
```

Because the Query is not returning a Paginator.